pkgs/applications/editors/emacs/elisp-packages/manual-packages.nix

```diff
@@ -29,7 +29,9 @@ in
 
   el-easydraw = callPackage ./manual-packages/el-easydraw { };
 
-  elisp-ffi = callPackage ./manual-packages/elisp-ffi { };
+  elisp-ffi = callPackage ./manual-packages/elisp-ffi {
+    inherit (pkgs) libffi pkg-config;
+  };
```

Comment:
What is the motivation?  It works without this change.

Reply:
I am being overzealous here.
I prefer to be explicit about which packages came from external `pkgs`, while the elisp ones can be kept implicit.

We do not have a package like `emacsPackages.pkg-config` yet, but better safe than sorry.

Reply:
Actually, this change is semantically significant.  Before, arguments like `libffi` are from [`pkgs'`][1] which cannot be `override`ed.  With this change applied, arguments like `libffi` are from [`pkgs`][2] which can be `override`ed.  So theoretically this change can cause breakage if users `override` `pkgs` to an attrset not containing needed arguments like `libffi`.

An example of `override` which **does not** overrides `pkgs` can be fount at [emacs-overlay][3].

[1]: https://github.com/NixOS/nixpkgs/blob/7c3a071a4c1cbb4d5628d1abb2cad5fc01898740/pkgs/top-level/emacs-packages.nix#L55
[2]: https://github.com/NixOS/nixpkgs/blob/7c3a071a4c1cbb4d5628d1abb2cad5fc01898740/pkgs/top-level/emacs-packages.nix#L56
[3]: https://github.com/nix-community/emacs-overlay/blob/7164cf7ca4318ffad776682ef8cf8aa9e24a1216/overlays/package.nix#L28

Reply:
After checking some other package sets in Nixpkgs, such as `qt6Packages`, I find that the convention is to use the old way.

Reply:
https://github.com/NixOS/nixpkgs/pull/329249